components/event-bus/cmd/mesh-namespaces-migration/subscription.go

```diff
@@ -0,0 +1,236 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kneventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	kneventingclientset "knative.dev/eventing/pkg/client/clientset/versioned"
+	"knative.dev/pkg/apis"
+	knv1alpha1 "knative.dev/pkg/apis/v1alpha1"
+
+	kymaeventingv1alpha1 "github.com/kyma-project/kyma/components/event-bus/apis/eventing/v1alpha1"
+	kymaeventingclientset "github.com/kyma-project/kyma/components/event-bus/client/generated/clientset/internalclientset"
+)
+
+// CloudEvents attributes
+const (
+	sourceAttr           = "source"
+	eventTypeAttr        = "type"
+	eventTypeVersionAttr = "eventtypeversion"
+)
+
+const (
+	defaultBrokerName = "default"
+	brokerLabel       = "eventing.knative.dev/broker"
+)
+
+type triggersByNamespaceMap map[string][]kneventingv1alpha1.Trigger
+type subscriptionsList []kymaeventingv1alpha1.Subscription
+
+// subscriptionMigrator performs migrations of Kyma Subscriptions to Knative Triggers.
+type subscriptionMigrator struct {
+	kymaClient    kymaeventingclientset.Interface
+	knativeClient kneventingclientset.Interface
+
+	subscriptions       subscriptionsList
+	triggersByNamespace triggersByNamespaceMap
+}
+
+// newSubscriptionMigrator creates and initializes a subscriptionMigrator.
+func newSubscriptionMigrator(kymaClient kymaeventingclientset.Interface,
+	knativeClient kneventingclientset.Interface, namespaces []string) (*subscriptionMigrator, error) {
+
+	m := &subscriptionMigrator{
+		kymaClient:    kymaClient,
+		knativeClient: knativeClient,
+	}
+
+	if err := m.populateSubscriptions(namespaces); err != nil {
+		return nil, err
+	}
+	if err := m.populateTriggers(namespaces); err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+// populateSubscription populates the local list of Kyma Subscriptions.
+func (m *subscriptionMigrator) populateSubscriptions(namespaces []string) error {
+	var kymaSubscriptions []kymaeventingv1alpha1.Subscription
+
+	for _, ns := range namespaces {
```

Comment:
rather then iterating through namespaces and finding kyma-subscriptions, does it makes sense to use SharedInformerFactory.Listers() this might give you cluster wide resources, right?

Reply:
An informer does the same: it calls List() and puts the results in a local cache. The difference is that it keeps watching for changes afterwards, which we don't need. I think an Informer is not what you want to use here.

Reply:
@anishj0shi  Thanks for the feedback
@antoineco WHat do you think?